providers/duo/v4.go

```diff
@@ -0,0 +1,230 @@
+package duo
```

Comment:
This is the meat of the new Duo OIDC implementation. It is _very_ similar to the old one, it's just slightly less complex. 

All URLs here are resolved relative to the BaseURL in DuoV4.